### ISSUE TITLE
Improve top-level error handling

### ISSUE THREAD

**Issue by kflansburg** [cloudflare/workers-rs#526]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Should we just accept `anyhow::Error` or something else generic?

**Comment by DylanRJohnston** [comment]
I'd just like to throw in my 5 cents while you're considering error handling. With the new axum support, the lack of implementation of `IntoResponse` for `worker::Error` presents a bit of an awkward usability problem when working with `axum`.

I currently have this newtype wrapper to work around and preserve the ergonomics of the `?` operator.

```rs
pub struct ErrWrapper(worker::Error);

impl From<worker::Error> for ErrWrapper {
    fn from(value: worker::Error) -> Self {
        ErrWrapper(value)
    }
}

impl IntoResponse for ErrWrapper {
    fn into_response(self) -> axum::response::Response {
        (StatusCode::INTERNAL_SERVER_ERROR, self.0.to_string()).into_response()
    }
}

pub type Result<T> = StdResult<T, ErrWrapper>;
```

**Comment by Jasper-Bekkers** [comment]
> Should we just accept anyhow::Error or something else generic?

Can't it just take a `Result<T, Box<dyn std::error::Error>>`?

**Comment by kflansburg** [comment]
Yes, I believe the current consensus is that the return type is something like that where `T: Into<web_sys::Response>`.
